deliveries/Bright_Disputes-Milestone_2.md

```diff
@@ -0,0 +1,18 @@
+# Milestone 2 Delivery :mailbox:
+
+* **Application Document:** https://github.com/bright/Ecosystem-Grants/blob/6a756f94bca60bc3d3bcbed64baf11675a5b1eca/applications/Bright_Disputes.md
+
+**Context** (optional)
+
+Bright Disputes is a dApp for solving disputes. Our goal is to create a smart contract (written in ink!) which could be used to raise and solve disputes on the Substrate-based blockchains. We would like to leverage a Zk-Snarks to accomplish privacy of voting.
+
+**Deliverables**
+
+| Number | Deliverable | Link | Notes |
+| -----: | ----------- |----------- | ------------- |
+| **0a.** | License |[LICENSE](https://github.com/bright/bright-disputes/blob/main/LICENSE) | MIT |
+| **0b.** | Documentation | [SHOWCASE](https://github.com/bright/bright-disputes/blob/main/doc/README.md) / [SHOWCASE_CLI](https://github.com/bright/bright-disputes/blob/main/doc/README_CLI.md) / [README](https://github.com/bright/bright-disputes/blob/main/README.md) | We will provide **inline documentation** of the code. |
+| **0c.** | Testing and Testing Guide | [README](https://github.com/bright/bright-disputes/blob/main/README.md) | Core functions will be fully covered by comprehensive unit tests to ensure functionality and robustness. In the guide, we will describe how to run these tests. |
+| **0d.** | Docker | [Dockerfile](https://github.com/bright/bright-disputes/blob/main/docker/Dockerfile) | Dockerfile that can be used to test all the functionality delivered with this milestone. |
+| 1. | CLI | [README](https://github.com/bright/bright-disputes/blob/main/cli/README.md) | Created a CLI for calling smart contract messages.
+| 2. | Library | [README](https://github.com/bright/aleph-node/blob/main/relations/ark/README.md) | We have created `vote` and `verdict` relations in the `liminal-ark-relations` library.
```

Comment:
Two questions here:
1. Can you give me a tl;dr of what R1CS is?
2. It looks like the linked code is part of a forked `aleph-node`, is this custom node required for your smart contracts to work?

If it is, are there any plans for you to upstream this work?

Reply:
Hello @HCastano,

let me answer on you questions:

1. R1CS stands for 'Rank-1 Constraint System,' and it is used as a mathematical representation of the problem behind zk-snark. In our case, one of the milestones' goals was to create two separate R1CS for voting and issuing the verdict. Thanks to the zk-snarks, we can achieve anonymity in voting and counting the votes.
2. We forked the 'aleph-node' because we need to extend the 'liminal-ark-relations' library (which is part of the 'aleph-node' repository) with our R1CS and also extend the 'cliain,' which is a command-line tool used to communicate with the aleph-node. Currently, we are not planning to merge those changes.